src/http/api/views/reply.chain.view.integration.test.ts

```diff
@@ -0,0 +1,168 @@
+
+    return {
+        ancestors,
+        post,
+        replies,
+        chains,
+    };
+}
+
+describe('ReplyChainView', () => {
+    let db: Knex;
+    let fixtureManager: FixtureManager;
+
+    beforeAll(async () => {
+        db = await createTestDb();
+        fixtureManager = createFixtureManager(db);
+    });
+
+    beforeEach(async () => {
+        await fixtureManager.reset();
+    });
+
+    describe('getAncestors', () => {
+        it('should return the paginated ancestors of a post', async () => {
+            const { ancestors, post } = await setupPosts(fixtureManager);
+
+            const replyChainView = new ReplyChainView(db);
+            // @ts-expect-error Property 'getAncestors' is private and only accessible within class 'ReplyChainView'
+            const result = await replyChainView.getAncestors(
+                post.author.id,
+                post.apId,
+            );
+
+            expect(result).toHaveLength(ReplyChainView.MAX_ANCESTOR_DEPTH);
+            expect(result[0].post_ap_id).toBe(
+                ancestors[ancestors.length - ReplyChainView.MAX_ANCESTOR_DEPTH]
+                    .apId.href,
+            );
+
+            // @ts-expect-error Property 'getAncestors' is private and only accessible within class 'ReplyChainView'
+            const remainingAncestors = await replyChainView.getAncestors(
+                post.author.id,
+                new URL(result[0].post_ap_id),
+            );
+
+            expect(remainingAncestors).toHaveLength(5);
+            expect(remainingAncestors[0].post_ap_id).toBe(
+                ancestors[0].apId.href,
+            );
+        });
+    });
+
+    describe('getChildren', () => {
+        it('should return the paginated children of a post with replies chain', async () => {
+            const { ancestors, post, replies, chains } =
+                await setupPosts(fixtureManager);
+
+            const replyChainView = new ReplyChainView(db);
+            // @ts-expect-error Property 'getChildren' is private and only accessible within class 'ReplyChainView'
+            const children = await replyChainView.getChildren(
+                post.author.id,
+                post.id!,
+            );
+
+            const expectedResults: string[] = [];
+
+            for (let i = 0; i < ReplyChainView.MAX_CHILDREN_COUNT + 1; i++) {
+                const child = replies[i];
+                expectedResults.push(child.apId.href);
+                const chain = chains[i];
+                const expectedChain = chain
+                    .slice(0, ReplyChainView.MAX_CHILDREN_DEPTH + 1)
+                    .map((post) => post.apId.href);
+                expectedResults.push(...expectedChain);
+            }
+
+            const resultIds = children.map((c) => c.post_ap_id);
+
+            expect(resultIds).toEqual(expectedResults);
+        });
+    });
+
+    describe('getReplyChain', () => {
+        it('should return the reply chain for a post', async () => {
```

Comment:
This test is a little shit for now - but the functionality is covered in the above two tests pretty extensively IMO